Source/Csla/ReadOnlyBase.cs

```diff
@@ -1520,8 +1521,16 @@ protected void MarkBusy()
     [EditorBrowsable(EditorBrowsableState.Advanced)]
     protected void MarkIdle()
     {
-      _isBusy = false;
-      OnBusyChanged(new BusyChangedEventArgs("", false));
+      int updatedValue = Interlocked.Decrement(ref _isBusy);
+      if (updatedValue < 0)
+      {
+        _ = Interlocked.CompareExchange(ref _isBusy, 0, updatedValue);
+        updatedValue = 0;
```

Comment:
That's not correct. With this you would trigger a PropertyChanged even though the state did not change. Keep the old updated value for the rest of the method. That's fine and desired.

Reply:
Fixed